artifactory/utils/container/image.go

```diff
@@ -55,12 +58,86 @@ func (image *Image) validateTag() error {
 
 // Get image base name by removing the prefixed registry hostname and the tag.
 // e.g.: https://my-registry/docker-local/hello-world:latest. -> hello-world
-func (image *Image) GetImageBaseName() (string, error) {
-	imageName, err := image.GetImageBaseNameWithTag()
+func (image *Image) GetImageShortName() (string, error) {
+	imageName, err := image.GetImageShortNameWithTag()
 	if err != nil {
 		return "", err
 	}
 	tagIndex := strings.LastIndex(imageName, ":")
+	if tagIndex != -1 {
+		return imageName[:tagIndex], nil
+	}
+	return imageName, nil
+}
 
-	return imageName[:tagIndex], nil
+func (image *Image) GetImageShortNameWithTag() (string, error) {
+	imageName, err := image.GetImageLongNameWithTag()
+	if err != nil {
+		return "", err
+	}
+	indexOfSlash := strings.LastIndex(imageName, "/")
+	if indexOfSlash != -1 {
+		return imageName[indexOfSlash+1:], nil
+
+	}
+	return imageName, nil
+}
+
+func (image *Image) GetImageTag() (string, error) {
+	imageName, err := image.GetImageLongNameWithTag()
+	if err != nil {
+		return "", err
+	}
+	tagIndex := strings.Index(imageName, ":")
+	return imageName[tagIndex+1:], nil
+}
+
+func (image *Image) GetRegistry() (string, error) {
+	if err := image.validateTag(); err != nil {
+		return "", err
+	}
+	indexOfLastSlash := strings.Index(image.name, "/")
+	return image.name[:indexOfLastSlash], nil
+}
+
+// Return the repository name of the pulled/pushed an image from Artifactory.
```

Comment:
Return the repository name of the pulled/pushed an image from Artifactory.
-->
Returns the physical Artifactory repository name of the pulled/pushed image, by reading a responso header from Artifactory.